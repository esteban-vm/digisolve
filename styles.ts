import { css } from '@emotion/react'

export const globalStyles = css`
  @import url('/css/fluid.css');
  @import url('/css/normalize.min.css');

  @font-face {
    font-family: 'Work Sans';
    src: local('Work Sans'), url('/fonts/WorkSans-VariableFont_wght.ttf');
  }

  :root {
    --color-primary: #ed07b7;
    --color-dark-primary: #bd0592;
    --color-secondary: #6920b0;
    --color-tertiary: #acdaff;

    --color-white: #fff;
    --color-grey: #555;
    --color-black: #000;

    --color-light: #f4f4f4;
    --color-lighter: #fcfcfc;

    --color-overlay1: #0000004d;
    --color-overlay2: #000000b3;
    --color-overlay3: #00000080;
    --color-overlay4: #000000cc;

    --color-shadow: #efefefef;
  }

  html,
  body {
    background-color: var(--color-white);
    color: var(--color-black);
    text-rendering: optimizeLegibility;
    font-size: 16px;
    font-family: 'Work Sans', Arial, sans-serif;
    font-weight: 300;
  }

  .row {
    max-width: 1140px;
    margin: 0 auto;
  }

  .clear_fix {
    zoom: 1;

    ::after {
      content: '';
      display: table;
      clear: both;
      visibility: hidden;
    }
  }
`

export * from '@emotion/react'
export { default as styled } from '@emotion/styled'
