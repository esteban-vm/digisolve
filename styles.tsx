import { Global, css } from '@emotion/react'

const styles = css`
  @import url('/css/fluid.css');
  @import url('/css/normalize.min.css');

  @font-face {
    font-family: 'Work Sans';
    src: local('Work Sans'), url('/fonts/WorkSans-VariableFont_wght.ttf');
  }

  :root {
    --primary-color: #ed07b7;
    --primary-color-dark: #bd0592;
    --secondary-color: #6920b0;
    --light-color: #acdaff;
    --dark-color: #555;
  }

  html,
  body {
    background-color: #fff;
    color: #000;
    text-rendering: optimizeLegibility;
    font-size: 16px;
    font-family: 'Work Sans', Arial, sans-serif;
    font-weight: 300;
  }

  .row {
    max-width: 1140px;
    margin: 0 auto;
  }
`

export const Globals = () => <Global styles={styles} />
export * from '@emotion/react'
export { default as styled } from '@emotion/styled'
