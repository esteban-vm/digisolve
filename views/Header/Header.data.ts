import type { NavLinkProps } from './NavLink'
import { uuid } from '@/helpers'

const links: NavLinkProps[] = [
  {
    id: uuid(),
    text: 'Home',
    link: '#',
  },
  {
    id: uuid(),
    text: 'Contact',
    link: 'contact',
  },
  {
    id: uuid(),
    text: 'About',
    link: 'about',
  },
]

export default links
