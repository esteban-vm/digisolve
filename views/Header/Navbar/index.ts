export { default, navLinks } from './Navbar'
