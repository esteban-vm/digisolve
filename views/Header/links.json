[
  {
    "text": "Home"
  },
  {
    "text": "Contact"
  },
  {
    "text": "About"
  }
]
