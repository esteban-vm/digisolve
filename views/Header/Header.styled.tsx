import { styled } from '@/styles'

export const Wrapper = styled.header`
  background-image: linear-gradient(rgba(0, 0, 0, 0.3), rgba(0, 0, 0, 0.7)), url('/img/header-img.jpg');
  background-size: cover;
  background-position: center;
  height: 100vh;
  position: relative;

  & h1 {
    font-size: 500%;
    color: #fff;
    font-weight: 400;
    margin-bottom: 20px;
  }

  & h2 {
    font-size: 200%;
    color: #fff;
    font-weight: 400;
  }
`

export const Box = styled.div`
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  text-align: center;
`

export const Text = styled.div`
  margin-bottom: 100px;
`
