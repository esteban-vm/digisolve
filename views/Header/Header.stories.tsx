import type { Meta } from '@/stories'
import Header from './Header'

export default {
  component: Header,
  title: 'Views/Header',
  parameters: {
    backgrounds: {
      disable: true,
    },
  },
} as Meta<typeof Header>

export const View = () => <Header />
