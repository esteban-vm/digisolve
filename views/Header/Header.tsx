import type { Component } from '@/types'
import { Button } from '@/components'
import { styled } from '@/styles'
import Navbar from './Navbar'

const HeaderComponent: Component = (props) => {
  return (
    <header {...props}>
      <Navbar />
      <div className='header__box'>
        <div className='header__titles'>
          <h1 className='header__heading'>Digital agency</h1>
          <h2 className='header__subheading'>The one stop for all your digital solutions</h2>
        </div>
        <Button text='Get a quote today!' isFull />
      </div>
    </header>
  )
}

const Header = styled(HeaderComponent)`
  background-image: linear-gradient(var(--color-overlay1), var(--color-overlay2)), url('/img/header-img.jpg');
  background-size: cover;
  background-position: center;
  height: 100vh;
  position: relative;
  color: var(--color-white);
  text-transform: capitalize;

  .header {
    &__box {
      position: absolute;
      top: 50%;
      left: 50%;
      transform: translate(-50%, -50%);
      text-align: center;
    }

    &__titles {
      margin-bottom: 100px;
    }

    &__heading {
      font-size: 500%;
      font-weight: 400;
      margin-bottom: 20px;
    }

    &__subheading {
      font-size: 200%;
      font-weight: 400;
    }
  }
`

export default Header
