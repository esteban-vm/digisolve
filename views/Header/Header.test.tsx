import { render, cleanup, screen, create } from '@/tests'
import { mediaQuery } from '@/styles'
import Header from './Header'

describe('ðŸ§ª HEADER:', () => {
  describe('display tests:', () => {
    /** Header title */
    const name = /^digital agency$/i

    beforeEach(() => {
      render(<Header />)
    })

    afterEach(cleanup)

    it('should be accessible', () => {
      const header = screen.getByRole('banner', { name })
      expect(header).toBeInTheDocument()
    })

    describe('should be displayed:', () => {
      it('the navbar', () => {
        const navbar = screen.getByRole('navigation')
        expect(navbar).toBeInTheDocument()
        expect(navbar).toBeVisible()
      })

      it('the title', () => {
        const title = screen.getByRole('heading', { name, level: 1 })
        expect(title).toBeInTheDocument()
        expect(title).toBeVisible()
      })

      it('the subtitle', () => {
        const subtitle = screen.getByRole('heading', { name: /all your digital solutions$/i, level: 2 })
        expect(subtitle).toBeInTheDocument()
        expect(subtitle).toBeVisible()
      })

      it('the button', () => {
        const button = screen.getByRole('button', { name: /^get a quote today\!$/i })
        expect(button).toBeInTheDocument()
        expect(button).toBeVisible()
      })
    })
  })

  describe('style tests:', () => {
    let tree: ReturnType<typeof create>

    beforeEach(() => {
      tree = create(<Header />)
    })

    it('should render with correct styles', () => {
      expect(tree).toMatchSnapshot()
    })

    it('should have correct styles on small screen devices', () => {
      expect(tree).toHaveStyleRule('margin-bottom', '50px', { target: '.titles', media: mediaQuery('sm') })
    })
  })
})
