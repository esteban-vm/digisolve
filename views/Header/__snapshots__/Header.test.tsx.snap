// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ðŸ§ª HEADER: style tests: should render with correct styles 1`] = `
.emotion-0 {
  background-image: linear-gradient(var(--color-overlay1), var(--color-overlay2)),url('/img/header-img.jpg');
  -webkit-background-size: cover;
  background-size: cover;
  -webkit-background-position: center;
  background-position: center;
  height: 100vh;
  position: relative;
  color: var(--color-white);
  text-transform: capitalize;
}

.emotion-0 .header__box {
  position: absolute;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%, -50%);
  -moz-transform: translate(-50%, -50%);
  -ms-transform: translate(-50%, -50%);
  transform: translate(-50%, -50%);
  text-align: center;
}

.emotion-0 .header__titles {
  margin-bottom: 100px;
}

.emotion-0 .header__heading {
  font-size: 500%;
  font-weight: 400;
  margin-bottom: 20px;
}

.emotion-0 .header__subheading {
  font-size: 200%;
  font-weight: 400;
}

.emotion-1 .navbar__image {
  height: 120px;
  width: auto;
  float: left;
}

.emotion-1 .navbar__list {
  list-style: none;
  float: right;
  margin-top: 70px;
}

.emotion-1 .navbar__item {
  display: inline-block;
  margin-right: 40px;
}

.emotion-1 .navbar__link:link,
.emotion-1 .navbar__link:visited {
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
  color: var(--color-white);
  outline: none;
  -webkit-transition: border-bottom 100ms;
  transition: border-bottom 100ms;
}

.emotion-1 .navbar__link:hover,
.emotion-1 .navbar__link:active {
  border-bottom: 2px solid var(--color-primary);
}

.emotion-2 {
  display: inline-block;
}

.emotion-2 .btn {
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: capitalize;
  padding: 10px 30px;
  border-radius: 200px;
  font-weight: 500;
  outline: none;
  -webkit-transition: background-color 500ms;
  transition: background-color 500ms;
  border: 2px solid var(--color-primary);
  background-color: var(--color-primary);
  color: var(--color-white);
}

.emotion-2 .btn:hover,
.emotion-2 .btn:active {
  background-color: var(--color-dark-primary);
}

<header
  className="emotion-0"
>
  <nav
    className="emotion-1"
  >
    <img
      alt="Digisolve logo"
      className="navbar__image"
      data-nimg="1"
      decoding="async"
      height={40}
      loading="lazy"
      onError={[Function]}
      onLoad={[Function]}
      src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
      srcSet="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
      style={
        {
          "color": "transparent",
        }
      }
      width={40}
    />
    <ul
      className="navbar__list"
    >
      <li
        className="navbar__item"
      >
        <a
          className="navbar__link"
          href="#"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
        >
          Home
        </a>
      </li>
      <li
        className="navbar__item"
      >
        <a
          className="navbar__link"
          href="#"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
        >
          Contact
        </a>
      </li>
      <li
        className="navbar__item"
      >
        <a
          className="navbar__link"
          href="#"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
        >
          About
        </a>
      </li>
    </ul>
  </nav>
  <div
    className="header__box"
  >
    <div
      className="header__titles"
    >
      <h1
        className="header__heading"
      >
        Digital agency
      </h1>
      <h2
        className="header__subheading"
      >
        The one stop for all your digital solutions
      </h2>
    </div>
    <span
      className="emotion-2"
    >
      <a
        className="btn"
        href="#"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
        role="button"
      >
        Get a quote today!
      </a>
    </span>
  </div>
</header>
`;
